app/Models/Contact/Contact.php

```diff
@@ -162,6 +163,26 @@ public function account()
         return $this->belongsTo(Account::class);
     }
 
+    /**
+     * Get the address book associated with the contact.
+     *
+     * @return BelongsTo
+     */
+    public function addressBook()
+    {
+        return $this->belongsTo(AddressBook::class);
+    }
+
+    /**
+     * Get the list of contacts from the same address book as this contact.
+     *
+     * @return HasMany|null
+     */
+    public function addressBookContacts()
```

Comment:
I don't think this is a good name.

To reduce complexity we should have simple names. I think, in this case, what are you trying to achieve? To get the contacts from the same address book.
So it could be `similarContactsInAddressBook()` or `sameAddressBookContacts()`.